### PR TITLE
feat: Webpack Memory Cache

### PR DESCRIPTION
**Pull request by bsokol-wl**
Enables Webpack's in-memory cache for dev builds

### CODE REVIEW

File: packages/webpack-configs/src/build.ts [hunk 1]
```diff
@@ -153,37 +145,7 @@ export function defineBuildConfig(swcConfig: SwcConfig, options: DefineBuildConf
...
-            resolve: {
-                hash: true,
-                timestamp: true
-            },
-            resolveBuildDependencies: {
-                hash: true,
-                timestamp: true
-            }
-        } : undefined,
+        cache: false,
```
Comment (patricklafrance) [inline packages/webpack-configs/src/build.ts:148]: According to webpack documentation, cache is disabled by default in production so you shouldn't need to set the option to `false`: https://reactrouter.com/en/main

File: packages/webpack-configs/src/dev.ts [hunk 2]
```diff
@@ -134,14 +128,8 @@ export function defineDevConfig(swcConfig: SwcConfig, options: DefineDevConfigOp
...
-            cacheDirectory: cacheDirectory,
-            // Took from https://webpack.js.org/configuration/cache/#cachebuilddependencies.
-            buildDependencies: {
-                config: [__filename]
-            }
+            type: "memory",
+            maxGenerations: 1
         },
         // (ACTUALLY NOT FIXING ANYTHING AT THE MOMENT)
         // Fixes caching for environmental variables using the DefinePlugin by forcing
```
Comment (patricklafrance) [inline packages/webpack-configs/src/dev.ts:135]: Is the snapshot section still required? If you prefer, this question can be answered after the PR is merge.

File: packages/webpack-configs/tests/build.test.ts [hunk 3]
```diff
@@ -179,23 +179,6 @@ test("when cache is enabled, the cache configuration is included", () => {
     expect(result.cache).toBeDefined();
```
Comment (patricklafrance) [inline packages/webpack-configs/tests/build.test.ts:179]: You should also remove this test.

File: packages/webpack-configs/src/build.ts (hunk 1 above)
Comment (bsokol-wl) [reply packages/webpack-configs/src/build.ts:148]: I'll remove this line

File: packages/webpack-configs/tests/build.test.ts (hunk 3 above)
Comment (bsokol-wl) [reply packages/webpack-configs/tests/build.test.ts]: removed

File: packages/webpack-configs/src/dev.ts (hunk 2 above)
Comment (bsokol-wl) [reply packages/webpack-configs/src/dev.ts:135]: Technically there's nothing wrong with keeping them and the builds will work fine as-is. But I'm looking at the default values, and I believe the addition of the extra hash checks in some of these would slow the rebuilds (probably not by much though). Regardless, I think it's best to remove these and let the defaults take over. I'll update the branch.

**Review by patricklafrance — APPROVED** [review]

**Pull request merged**